buildSrc/src/main/java/com/osacky/flank/gradle/YamlWriter.kt

```diff
@@ -106,6 +128,44 @@ internal class YamlWriter {
     config.resultsDir?.let {
       appendln("  results-dir: $it")
     }
+
+    config.testRunnerClass?.let {
+      appendln("  test-runner-class: $it")
+    }
+
+    config.numUniformShards?.let {
+      appendln("  num-uniform-shards: $it")
+    }
+
+    if (config.clientDetails.isNotEmpty()) {
+      appendln("  client-details:")
+      config.clientDetails.forEach {
+        appendln("    ${it.key}: ${it.value}")
+      }
+    }
+
+    if (config.otherFiles.isNotEmpty()) {
+      appendln("  other-files:")
+      config.otherFiles.forEach {
+        appendln("    ${it.key}: ${it.value}")
+      }
+    }
+
+    config.networkProfile?.let {
+      appendln("  network-profile: $it")
+    }
+
+    config.roboScript?.let {
+      appendln("  robo-script: $it")
+    }
+
+    if (config.roboDirectives.isNotEmpty()) {
+      appendln("  robo-directives:")
+      config.roboDirectives.forEach {
+        val value = it.getOrNull(2).let { stringValue -> if (stringValue.isNullOrBlank()) "\"\"" else stringValue }
```

Comment:
if `it.getOrNull(2)` returns null, does it crash on `.let`? should it be `?.let`?

Reply:
Interesting fact it worked! But I think you are right.  I'll change it to be more robust